src/main/java/mike/controller/FrameController.java

```diff
@@ -20,19 +20,20 @@
 import mike.datastructures.Relationship;
 import mike.gui.Line;
 import mike.view.GUIView;
+import mike.view.ViewTemplate;
 
 public class FrameController {
 	
-	protected static void exitListener() throws HeadlessException {	
-		GUIView.getFrame().setDefaultCloseOperation(JFrame.DO_NOTHING_ON_CLOSE);
-		GUIView.getFrame().addWindowListener(new WindowAdapter()
+	protected static void exitListener(boolean changed, ViewTemplate view) throws HeadlessException {	
+	    ((GUIView) view).getFrame().setDefaultCloseOperation(JFrame.DO_NOTHING_ON_CLOSE);
+	    ((GUIView) view).getFrame().addWindowListener(new WindowAdapter()
 	    {
 	        //@Override
 	        public void windowClosing(WindowEvent e)
 	        {	
-	        	if(Controller.getChanged()){
+	        	if(changed){
```

Comment:
Related to Stefan's comment:

Exit listener should take in Controller instead, so it can get the value of changed dynamically, rather than whenever this listener was created.
The if statement on line 34 should then check:
if(control.getChanged()) to see if it is true.

Reply:
Gotcha. So does that change need to be made and pushed then it should go back to working like before?

Reply:
I believe so. I can do that and push it real quick - got sidetracked with a physics lab that I have since given up on for now